src/diem/offchain/funds_pull_pre_approval_command_state.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) The Diem Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import typing
+
+from .types import FundPullPreApprovalStatus
+
+Status = str
+
+_transitions: typing.Dict[Status, typing.List[Status]] = {
+    FundPullPreApprovalStatus.pending: [
+        FundPullPreApprovalStatus.pending,
+        FundPullPreApprovalStatus.valid,
+        FundPullPreApprovalStatus.rejected,
+        FundPullPreApprovalStatus.closed,
+    ],
+    FundPullPreApprovalStatus.valid: [
+        FundPullPreApprovalStatus.valid,
+        FundPullPreApprovalStatus.closed,
+    ],
+    FundPullPreApprovalStatus.rejected: [
+        FundPullPreApprovalStatus.rejected,
+    ],
+    FundPullPreApprovalStatus.closed: [
+        FundPullPreApprovalStatus.closed,
+    ],
+}
+
+_initial_states: typing.List[Status] = [
+    FundPullPreApprovalStatus.pending,
+    FundPullPreApprovalStatus.valid,
```

Comment:
Why valid is a valid initial state?

Reply:
Because there are flows (these are actually expected to be the more common flows), where the first message is sent by the paying side. In this case the first message immediately has the status "valid" or "rejected".